src/scanpy/preprocessing/_pca/_dask_sparse.py

```diff
@@ -0,0 +1,183 @@
+
+        self.explained_variance_ratio_ = self.explained_variance_ratio_[
+            : self.n_components_
+        ]
+        return self
+
+    def transform(self: PCASparseFit, x: DaskArray) -> DaskArray:
+        if TYPE_CHECKING:
+            import dask.array.core as da
+        else:
+            import dask.array as da
+
+        def transform_block(
+            x_part: CSMatrix,
+            mean_: NDArray[np.floating],
+            components_: NDArray[np.floating],
+        ):
+            pre_mean = mean_ @ components_.T
+            mean_impact = np.ones((x_part.shape[0], 1)) @ pre_mean.reshape(1, -1)
+            return (x_part @ components_.T) - mean_impact
+
+        return da.map_blocks(
+            transform_block,
+            x,
+            mean_=self.mean_,
+            components_=self.components_,
+            chunks=(x.chunks[0], self.n_components_),
+            meta=np.zeros([0], dtype=x.dtype),
+            dtype=x.dtype,
+        )
+
+    def fit_transform(self, x: DaskArray, y: DaskArray | None = None) -> DaskArray:
+        if y is None:
+            y = x
+        return self.fit(x).transform(y)
+
+
+@overload
+def _cov_sparse_dask(
+    x: DaskArray, *, return_gram: Literal[False] = False, dtype: DTypeLike | None = None
+) -> tuple[NDArray[np.floating], NDArray[np.floating]]: ...
+@overload
+def _cov_sparse_dask(
+    x: DaskArray, *, return_gram: Literal[True], dtype: DTypeLike | None = None
+) -> tuple[NDArray[np.floating], NDArray[np.floating], NDArray[np.floating]]: ...
+def _cov_sparse_dask(
+    x: DaskArray, *, return_gram: bool = False, dtype: DTypeLike | None = None
+) -> (
+    tuple[NDArray[np.floating], NDArray[np.floating], NDArray[np.floating]]
+    | tuple[NDArray[np.floating], NDArray[np.floating]]
+):
+    """\
+    Computes the covariance matrix and row/col means of matrix `x`.
+
+    Parameters
+    ----------
+
+    x
+        A sparse matrix
+    return_gram
+        If `True`, the gram matrix will be returned and a copy will be created
+        to store the results of the covariance,
+        while if `False`, the local gram matrix result will be overwritten.
+
+    Returns
+    -------
+
+    :math:`\\cov(X, X)`
+        The covariance matrix of `x` in the form :math:`\\cov(X, X) = \\E(XX) - \\E(X)\\E(X)`.
+    :math:`\\gram(X, X)`
+        When return_gram is `True`, the gram matrix of `x` in the form :math:`\\frac{1}{n} X.T \\dot X`.
+    :math:`\\mean(X)`
+        The row means of `x`.
+    """
+    if TYPE_CHECKING:
+        import dask.array.core as da
+        import dask.base as dask
+    else:
+        import dask
+        import dask.array as da
+
+    if dtype is None:
+        dtype = np.float64 if np.issubdtype(x.dtype, np.integer) else x.dtype
+    else:
+        dtype = np.dtype(dtype)
+
+    def gram_block(x_part: CSMatrix):
+        gram_matrix: CSMatrix = x_part.T @ x_part
+        return gram_matrix.toarray()[None, ...]  # need new axis for summing
```

Comment:
Did we not consider speedups @Intron7 ?

Reply:
I think this is quite well optimzed atm. I have to do tests on how multicore numbakernels work within dask. If it turns out to be faster we can still update.

Reply:
We do we need multicore kernels?  Dask workers are roughly mappable to cores.  Do we think the algorithm would be so different from optimized densification procedures on the GPU?